packages/eslint-plugin/src/rules/unbound-method.ts

```diff
@@ -108,6 +120,21 @@ function isSafeUse(node: TSESTree.Node): boolean {
       return true;
 
     case AST_NODE_TYPES.CallExpression:
+      if (
+        parent.callee.type === AST_NODE_TYPES.MemberExpression &&
+        parent.callee.property.type === AST_NODE_TYPES.Identifier
+      ) {
```

Comment:
What I really would like to do, is to check if callee for ex. `nodes` is an `Array`. Is this possible?

Reply:
you should be able to get the type of the node, and ensure its type is an array.
I can't remember the flags exactly, so you might want to spend some time with the debugger in vscode.

Reply:
I've spent some time trying to play with the debugger to find any flags that might indicate `Array` type, but with no success for now. If you think we can't leave it like this, I'll spend more time and maybe find the better way.

Reply:
Digging back into this, sorry a lot going on. Finally taken time to tackle the PR queue.

Feel free to add this file into your PR, and use that API.
https://github.com/typescript-eslint/typescript-eslint/blob/naming-conventions/packages/eslint-plugin/typings/typescript.d.ts

The API is "internal", but it's stable - they just haven't gotten around to exposing their type relationship API yet (there's a whole discussion around why they haven't yet...)

You can then pass the typechecker into this function and call `checker.isArrayType(type)` (you'll also have to pass the type map in).

Also just as an FYI - this site is amazing for playing with TS's type API: https://ts-ast-viewer.com/
You can open the chrome dev tools, and interrogate the selected type via the global variables it maintains: `type`, `program` and `typeChecker`.